### ISSUE TITLE
Your Session has crashed due to unknown reason  

### ISSUE THREAD

**Issue by DarshanDeshpande** [tensorflow/tensorflow#35558]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Custom Code
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 2.1.0-rc1
- Python version: 3.6
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 
- GPU model and memory: Colab Notebook default GPU/CPU

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
I am trying to train a model using two separate sub-models, each using audio and image data respectively. I have created an input pipeline having (audio data,image data and binary labels). According to <a href="https://www.tensorflow.org/guide/keras/train_and_evaluate#training_evaluation_from_tfdata_datasets">this</a> guide, I normally passed the tf.data.Dataset to the model.fit() method and soon the colab notebook crashed, giving an "unknown error". 

**Describe the expected behavior**
Starting of the training process

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

```
filenames = [data_path]
raw_dataset = tf.data.TFRecordDataset(filenames)
raw_dataset

feature = {'train/image': tf.io.FixedLenFeature([], tf.string),
            'train/label': tf.io.FixedLenFeature([], tf.int64),
           'train/audio': tf.io.FixedLenFeature([], tf.string),}

def _parse_function(example_proto):
  features = tf.io.parse_single_example(example_proto, feature)
  audio_raw = tf.expand_dims(tf.io.decode_raw(features['train/audio'],tf.float32),axis=-1)
  image_raw = tf.io.decode_raw(features['train/image'],tf.float32)
  image = tf.reshape(image_raw,[5,500,500,3])
  labels =features['train/label']
  labels = tf.one_hot(labels,depth=2)
  return audio_raw,image,labels

dataset = raw_dataset.map(_parse_function)
dataset

BATCH_SIZE = 4
BUFFER_SIZE = 10000

dataset = dataset.cache().shuffle(BUFFER_SIZE).batch(BATCH_SIZE).repeat()
dataset
```
<br><br>

```
am = action_model((5,500,500,3))
afm = audio_and_final_model((105000,1))

combined = tf.keras.layers.Concatenate(axis=-1)([am.output,afm.output])
z = tf.keras.layers.Dense(32,activation='relu')(combined)
dropOut = tf.keras.layers.Dropout(0.3)(z)
out = tf.keras.layers.Dense(2,activation='tanh')(dropOut)
model = tf.keras.models.Model(inputs=[am.input,afm.input],outputs=out)
model.compile(loss='hinge', optimizer='adam', metrics=['accuracy'])

# print(model.summary())
EPOCHS=30
callbacks = [tf.keras.callbacks.ReduceLROnPlateau(verbose=1,monitor='loss',patience=2),tf.keras.callbacks.ModelCheckpoint('/content/drive/My Drive/Model.h5',monitor='loss',save_best_only=True,save_freq='epoch',save_weights_only=True),tf.keras.callbacks.TensorBoard()]
print("Starting training")
model.fit(dataset,epochs=EPOCHS,callbacks=callbacks)
print("Saving Model")
model.save("Model.h5")
```

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.
Logs- https://pastebin.com/BdXCRdvV

**Comment by DarshanDeshpande** [comment]
So I guess the Colab GPU couldn't handle the model and thats the reason it crashed. Changed my model architecture now and everything works well. Thanks! Closing this issue
